.github/workflows/cicd.yaml

```
@@ -5,53 +5,62 @@ on:
   pull_request:
     branches: [main]
 
+env:
+  LATEST_PY_VERSION: '3.14'
+
 jobs:
   test:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.9", "3.10", "3.11", "3.12", "3.13"]
+        python-version: ["3.9", "3.10", "3.11", "3.12", "3.13", "3.14"]
     timeout-minutes: 20
 
     steps:
       - name: Check out repository code
         uses: actions/checkout@v5
-
-      # Setup Python (faster than using Python container)
-      - name: Setup Python
-        uses: actions/setup-python@v6
+      
+      - name: Install uv
+        uses: astral-sh/setup-uv@v3
         with:
-          python-version: ${{ matrix.python-version }}
-
-      - name: Lint code
-        if: ${{ matrix.python-version == 3.13 }}
-        run: |
-          python -m pip install pre-commit
-          pre-commit run --all-files
-
-      - name: Install types
-        run: |
-          python -m pip install ./stac_fastapi/types[dev]
+          version: "0.9.*" 
+          enable-cache: true
+    
+      - name: Set up Python ${{ matrix.python-version }}
+        run: uv python install ${{ matrix.python-version }}
 
-      - name: Install core api
+      - name: Install dependencies
         run: |
-          python -m pip install ./stac_fastapi/api[dev]
+          uv sync --all-extras
 
-      - name: Install Extensions
+      - name: Lint code
+        if: ${{ matrix.python-version == env.LATEST_PY_VERSION }}
         run: |
-          python -m pip install ./stac_fastapi/extensions[dev]
+          uv run pre-commit run --all-files
 
-      - name: Test
-        run: python -m pytest -svvv
+      - name: Run tests
+        run: uv run pytest -svvv
         env:
           ENVIRONMENT: testing
 
   test-docs:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v5
-      - name: Test generating docs
-        run: make docs
+      - name: Checkout main
+        uses: actions/checkout@v5
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v3
+        with:
+          version: "0.9.*" 
+          enable-cache: true
+    
+      - name: Set up Python3
+        run: uv python install 3
+      
+      - name: Test docs
+        run: |
+          uvx --with-requirements requirements/requirements-docs.txt mkdocs build -f docs/mkdocs.yml 
```

If you move the docs dependencies into a `dependency-groups` called `docs` you can do this:
```suggestion
          uv run --group docs mkdocs build -f docs/mkdocs.yml 
```